### PR TITLE
[Merged by Bors] - Switch to workspace inherited properties

### PR DESCRIPTION
**Pull request by jedel1043**
This Pull Request switches our codebase to the brand new [workspace inherited keys](https://doc.rust-lang.org/cargo/reference/workspaces.html#the-package-table), which allows us to define common package options that are usable within each crate's Cargo.toml file.

It also allows to share dependency versions between crates, but I defined only shared versions for our workspace members. It would be a good follow-up to lift all the shared dependencies between crates into the global Cargo.toml.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -11,6 +11,22 @@ members = [
     "boa_examples",
 ]
 
+[workspace.package]
+edition = "2021"
+version = "0.15.0"
+rust-version = "1.60"
```
Comment (Razican) [inline Cargo.toml:17]: We might need to specify 1.64 here, since this is only available starting on Rust 1.64 (or cargo 1.64, in fact)

Comment (jedel1043) [reply Cargo.toml:17]: Yep!

**Comment by jedel1043** [comment]
Apparently dependabot doesn't have support for inherited dependencies (tracked in https://github.com/dependabot/dependabot-core/issues/5315), so lifting the common dependencies is blocked by that issue.

**Review by raskad — APPROVED** [review]
Looks good to me.
We should probably keep this change in mind for the 0.17 release, to see if it works correct with `cargo-workspaces publish`.

File: Cargo.toml [hunk 2]
```diff
@@ -11,6 +11,22 @@ members = [
...
+version = "0.15.0"
+rust-version = "1.64"
+authors = ["boa-dev"]
+repository = "https://github.com/boa-dev/boa"
+license = "Unlicense/MIT"
+description = "Boa is a Javascript lexer, parser and Just-in-Time compiler written in Rust. Currently, it has support for some of the language."
+
+[workspace.dependencies]
+boa_engine = { version = "0.15.0", path = "boa_engine" }
+boa_interner = { version = "0.15.0", path = "boa_interner" }
```
Comment (jasonwilliams) [inline Cargo.toml:25]: What happens if version is not specified?

Comment (jedel1043) [reply Cargo.toml:25]: Crates.io rejects publishing the crate

**Review by jasonwilliams — APPROVED** [review]

**Comment by jedel1043** [comment]
Should we push this into 0.16, just to see if `cargo-workspaces` work? We can rollback if there's any problem

**Comment by poliorcetics** [comment]
You won't have any rust-related dependency update from dependabot now I think since it doesn't yet know where to find the version number for a `dep.workspace = true` dependency, even if that dependency is local.

I opened a PR for it https://github.com/dependabot/dependabot-core/pull/5794, it should work but it's not merged yet.

It's probably possible to regenerate the docker container and use it locally to make the necessary changes but I have no idea how to do that. I'll gladly accept people testing it and telling me if it works has expected in a variety of configs though 👍

**Pull request closed without merging**